### PR TITLE
Let user create a new dashboard when adding a question to a dashboard

### PR DESCRIPTION
**Pull request by wtfleming**
Possible implementation for #1017

Up until recently I was working at Spot, and created enough dashboards that not having this might be my biggest pain point with Metabase. Wanted to scratch that itch and use it as an opportunity to learn more about React. If there is a better (or more idiomatic) way to approach this i'd love feedback.

It seems to work as I would expect it to when creating a new question, adding an existing question, or after pressing the back button in the browser.

(This obviously still needs design / UX improvements)

add-question-to-dashboard

Clicking on the button transitions the modal to this

create-dashboard
###### TODO
-  [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
  (unless it's a tiny documentation change).

### CODE REVIEW

File: frontend/src/metabase/components/AddToDashSelectDashModal.jsx [hunk 1]
```diff
@@ -62,6 +63,13 @@ export default class AddToDashSelectDashModal extends Component {
                         items={this.state.dashboards}
                         onClickItemFn={this.addToDashboard}
                     />
+                    <button
+                        className="Button Button--primary"
+                        onClick={() => this.setState({shouldCreateDashboard: true})}
```
Comment (tlrobinson) [inline frontend/src/metabase/components/AddToDashSelectDashModal.jsx]: Minor nitpick, we generally put spaces inside the brackets here, e.x. `{ shouldCreateDashboard: true }`

**Comment by tlrobinson** [comment]
@wtfleming Thanks for doing this. The code looks good.

@mazameli @kdoh Thoughts on the button positioning and button text?

Here's centered with pb2:

screenshot 2016-10-17 16 17 03

Centered with pb4:

screenshot 2016-10-17 16 21 23

Left aligned with pb4 px4

screenshot 2016-10-17 16 21 41

**Comment by mazameli** [comment]
I love, love, love this functionality. Thanks for this!

Regarding placement of the button, my preference would be this:
popup - pick dahsboard to add to copy

- I think a big blue button is too strong, since this is a secondary action
- I prefer the action to be at the top, so that if you have a long list of dashboards, you don't have to scroll to the bottom to get to it

@kdoh, do we have a link-with-icon or some such component that Will could use here?

**Comment by kdoh** [comment]
@mazameli I don't think we do currently but should just be a matter of having an Icon component inside a Link component with flex and align-center for class names.

**Comment by kdoh** [comment]
And as has been said before, props to @wtfleming.

**Comment by wtfleming** [comment]
Changed to look closer to preferences from @mazameli 

v2-add-to-dashboard

It's almost, but not quite there. Color on the add to new dashboard section is different (but it does turn blue when mousing over it).

A bigger issue is alignment, I learned a fair bit about CSS working on this, but couldn't get it quite right without adding it to the `SortableItemList` component - which I don't think is desirable since it's used in both `AddToDashSelectDashModal` and `AddToDashSelectQuestionModal` components.

Or to put it another way, you likely don't want the "Add to new dashboard" also included in this modal:

add-question

Maybe there's a way to make it work via CSS that i'm not aware of, or another approach I could take?

File: frontend/src/metabase/components/AddToDashSelectDashModal.jsx [hunk 2]
```diff
@@ -58,10 +60,21 @@ export default class AddToDashSelectDashModal extends Component {
                     title="Add Question to Dashboard"
                     closeFn={this.props.closeFn}
                 >
+                <div className="flex flex-column">
+                    <div
+                        className="text-brand-hover flex-align-right px4 cursor-pointer"
```
Comment (kdoh) [inline frontend/src/metabase/components/AddToDashSelectDashModal.jsx]: @wtfleming If you use the `link` class here you can remove `text-brand-hover` and get the link color behavior you're looking for for the action.

File: frontend/src/metabase/components/AddToDashSelectDashModal.jsx [hunk 3]
```diff
@@ -58,10 +60,21 @@ export default class AddToDashSelectDashModal extends Component {
                     title="Add Question to Dashboard"
                     closeFn={this.props.closeFn}
                 >
+                <div className="flex flex-column">
```
Comment (kdoh) [inline frontend/src/metabase/components/AddToDashSelectDashModal.jsx]: @wtfleming Try adding `align center` on this parent and I believe this should fix the alignment issue you mentioned.

**Comment by kdoh** [comment]
@wtfleming Left a couple comments that should hopefully help out with the issues you mentioned.

File: frontend/src/metabase/components/AddToDashSelectDashModal.jsx (hunk 3 above)
Comment (tlrobinson) [inline frontend/src/metabase/components/AddToDashSelectDashModal.jsx]: @kdoh do you mean `align-center`?

Comment (kdoh) [inline frontend/src/metabase/components/AddToDashSelectDashModal.jsx]: I do indeed

Comment (kdoh) [inline frontend/src/metabase/components/AddToDashSelectDashModal.jsx]: Hrm yeah that's not exactly what we want is it :P Let me pull your branch and try something and get back to you shortly.

Comment (wtfleming) [inline frontend/src/metabase/components/AddToDashSelectDashModal.jsx]: thanks @kdoh made the change and pushed it

**Comment by mazameli** [comment]
Really looking forward to getting this merged into the product, @wtfleming. Anything we can do to help? Appreciate your work on this!

**Comment by kdoh** [comment]
At my last check this looked good to merge whenever we're ready. Props to @wtfleming.

**Pull request merged**